File: Pascals_Triangle/pascals_triangle.java
import java.util.List;
import java.util.ArrayList;

class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<>();
        for (int i = 0; i < numRows; i++) {
            List<Integer> tmp = new ArrayList<>();
            tmp.add(1);
            for (int j = 1; j < i; j++) {
                int tmp2 = res.get(i - 1).get(j - 1) + res.get(i - 1).get(j);
                tmp.add(tmp2);
            }
            if (i > 0) {
                tmp.add(1);
            }
            res.add(tmp);
        }
        return res;
    }
}

File: Pascals_Triangle/pascals_triangle.py

class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        res = []
        for i in range(numRows):
            res.append([])
            res[i].append(1)
            for j in range(1,i):
                res[i].append(res[i - 1][j - 1] + res[i - 1][j])
            if i > 0:
                res[i].append(1)
        return res
